### ISSUE TITLE
Error migrating file relations: "foreign key constraint cannot be implemented"

### ISSUE THREAD

**Issue by william-roy** [directus-community/migration-tool#35]
First I was getting 400 bad request error. Then I tried the branch by @hakan-erdem-temiz based on his PR from a month ago. This got me past the 400 and migrated many of the relations until it hit  a new 500. I think this is the only collection that uses the 02M "Files" field where it seems to be snagging.

Here's the log from the script:
~~~
 > Migrating Schema
    √ Downloading Schema
    √ Creating Collections
    × Migrating Relations
      → Request failed with status code 500
    Migration Files
    Migrating Users
    Migrating Data
Error: Request failed with status code 500
    at createError (C:\Users\liamh\dev\directus-migration-tool-2\node_modules\axios\lib\core\createError.js:16:15)
    at settle (C:\Users\liamh\dev\directus-migration-tool-2\node_modules\axios\lib\core\settle.js:17:12)
    at IncomingMessage.handleStreamEnd (C:\Users\liamh\dev\directus-migration-tool-2\node_modules\axios\lib\adapters\http.js:260:11)
    at IncomingMessage.emit (events.js:387:35)
    at endReadableNT (internal/streams/readable.js:1317:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  config: {
    url: '/relations',
    method: 'post',
    data: '{"meta":{"many_collection":"projects_directus_files","many_field":"directus_files_id","many_primary":"id","one_collection":"directus_files","one_field":"projects","one_primary":"id","junction_field":"projects_id"},"field":"directus_files_id","collection":"projects_directus_files","related_collection":"directus_files","schema":null}',
~~~

...and here's the log from the v9 instance:

~~~
21:38:44 🚨 error alter table "projects_directus_files" add constraint "projects_directus_files_directus_files_id_foreign" foreign key ("directus_files_id") references "directus_files" ("id") - foreign key constraint "projects_directus_files_directus_files_id_foreign" cannot be implemented
error: alter table "projects_directus_files" add constraint "projects_directus_files_directus_files_id_foreign" foreign key ("directus_files_id") references "directus_files" ("id") - foreign key constraint "projects_directus_files_directus_files_id_foreign" cannot be implemented
    at Parser.parseErrorMessage (/home/node/.npm/_npx/ebcc468e2b0f2b61/node_modules/pg-protocol/dist/parser.js:287:98)
    at Parser.handlePacket (/home/node/.npm/_npx/ebcc468e2b0f2b61/node_modules/pg-protocol/dist/parser.js:126:29)
    at Parser.parse (/home/node/.npm/_npx/ebcc468e2b0f2b61/node_modules/pg-protocol/dist/parser.js:39:38)
    at TLSSocket.<anonymous> (/home/node/.npm/_npx/ebcc468e2b0f2b61/node_modules/pg-protocol/dist/index.js:11:42)
    at TLSSocket.emit (node:events:394:28)
    at addChunk (node:internal/streams/readable:312:12)
    at readableAddChunk (node:internal/streams/readable:287:9)
    at TLSSocket.Readable.push (node:internal/streams/readable:226:10)
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)
~~~
![image](https://user-images.githubusercontent.com/51927032/124838523-068c3c00-df55-11eb-9928-b82201441876.png)

Directus 8 v8.7.2 on MySQL 8
Directus 9 v9.0.0-rc83 on PostgreSQL 12

**Comment by moekify** [comment]
I'm having the same problem it seems. Failing to insert items due to foreign key constraints. Seemingly this is related to it trying to push relations for non-existent items (i.e. I add an item content, that references an item in another collection that hasn't been added yet) Hence, the adding of the content item fails :(

```
postgres_1  | 2021-07-10 12:06:28.517 UTC [72] ERROR:  insert or update on table "content" violates foreign key constraint "content_content_for_project_foreign"
postgres_1  | 2021-07-10 12:06:28.517 UTC [72] DETAIL:  Key (content_for_project)=(4) is not present in table "projects".
postgres_1  | 2021-07-10 12:06:28.517 UTC [72] STATEMENT:  insert into "content" ("captions", "content_for_project", "content_label", "created_by", "created_on", "distance_to_next", "hero_background_color", "id", "on_page", "sort", "status", "text", "text_layout", "type") values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14)
directus_1  | 12:06:28 🐛 Invalid foreign key in field "content_for_project".
directus_1  | 12:06:28 ✨ request completed POST 400 /items/content 19ms
```

The projects weren't actually imported when the content (that references the project) was added. 

@rijkvanzanten any smart ideas on how to navigate around this? 
Basically, we somehow need to first add all items without relation fields and then update all items with the relations.

**Comment by rijkvanzanten** [comment]
> Basically, we somehow need to first add all items without relation fields and then update all items with the relations.

That would be my suggestion as well. Import the items data first, and _then_ apply the relationships (eg foreign key constraints)
